src/components/fields/location-input-group/location-modal/location-search/location-search.tsx

```diff
@@ -0,0 +1,749 @@
+	displayText?: string;
+}
+
+export interface ILocationSearchProps {
+	id?: string;
+	onCancel: () => void;
+	onConfirm: () => void;
+	isSinglePanel: boolean;
+	singelPanelInputMode: TSinglePanelInputMode;
+	addressFieldPlaceholder?: string;
+	gettingCurrentLocation: boolean;
+	gettingCurrentLocationFetchMessage?: string;
+	locationListTitle?: string;
+	selectedAddressInfo: ILocationInputValues;
+	changeSelectedAddressInfo: (addressInfo: ILocationInputValues) => void;
+	pageSize?: number;
+	onOneMapError: () => void;
+	mustHavePostalCode?: boolean;
+	disableErrorPromptOnApp?: boolean;
+	reverseGeoCodeEndpoint?: string;
+	onGetLocationCallback: (lat?: number, lng?: number) => void;
+	onGetLocationError: (
+		geolocationPositionError?: GeolocationPositionError,
+		disableErrorPromptOnApp?: boolean
+	) => void;
+	onAddressChange: () => void;
+	showLocationModal: boolean;
+	mapPickedLatLng?: ILocationCoord;
+	didUserClickMap: boolean;
+	onClearInput: () => void;
+	formValues?: ILocationInputValues;
+	updateFormValues: (values: ILocationInputValues) => void;
+}
+
+const pagination = <T,>(array: T[], pageSize: number, pageNum: number) => {
+	return array.slice((pageNum - 1) * pageSize, pageNum * pageSize);
+};
+
+const fetchAddress = async (
+	query: string,
+	pageNum: number,
+	onSuccess?: (results: OneMapSearchResults) => void,
+	onFail?: (error: unknown) => void
+) => {
+	if (query) {
+		try {
+			const results = await OneMapService.searchByAddress({
+				searchValue: query,
+				getAddressDetails: OneMapBoolean.YES,
+				returnGeom: OneMapBoolean.YES,
+				pageNum: pageNum,
+			});
+			onSuccess?.(results);
+		} catch (error) {
+			onFail?.(error);
+		}
+	}
+};
+
+const debounceFetchAddress = debounce(fetchAddress, 500);
+
+const boldResultsWithQuery = (arr: OneMapSearchBuildingResult[], query: string) => {
+	const regex = new RegExp(query, "gi");
+	return arr.map((obj) => {
+		const newAddress = (obj.DISPLAY_ADDRESS || obj.ADDRESS).replace(regex, `<span class="keyword">${query}</span>`);
+		return {
+			...obj,
+			DISPLAY_ADDRESS: newAddress,
+		};
+	});
+};
+
+const cleanHtml = (html: string | undefined) => {
+	if (!html) return;
+	return {
+		__html: sanitizeHtml(html, {
+			allowedTags: ["span"],
+			allowedAttributes: {
+				span: ["class"],
+			},
+		}),
+	};
+};
+
+export const LocationSearch = ({
+	id = "location-search",
+	onCancel,
+	singelPanelInputMode,
+	isSinglePanel,
+	onConfirm,
+	addressFieldPlaceholder = "Street Name, Postal Code",
+	gettingCurrentLocation,
+	gettingCurrentLocationFetchMessage = "Getting current location...",
+	locationListTitle = "Select location",
+	selectedAddressInfo,
+	changeSelectedAddressInfo,
+	pageSize = 10,
+	onOneMapError,
+	mustHavePostalCode,
+	disableErrorPromptOnApp,
+	reverseGeoCodeEndpoint,
+	onGetLocationCallback,
+	onGetLocationError,
+	onAddressChange,
+	showLocationModal,
+	mapPickedLatLng,
+	didUserClickMap,
+	onClearInput,
+	formValues,
+	updateFormValues,
```

Comment:
- combine `singelPanelInputMode` and `isSinglePanel` since they are used in similar ways
- ~~changeSelectedAddressInfo~~ `onChangeSelectedAddressInfo`
- remove `pageSize` since it is always 10
- combine `onOneMapError` and `onGetLocationError` into `onError` and differentiate between them through arguments instead
- `disableErrorPromptOnApp` is likely not necessary
- `didUserClickMap` does not seem necessary, `mapPickedLatLng` will be set when user clicks the location picker
- try to differentiate similar-looking prop names
  - e.g. `changeSelectedAddressInfo` vs `onAddressChange`